tensorly/decomposition/parafac2.py

```diff
@@ -40,13 +40,18 @@ def initialize_decomposition(tensor_slices, rank, init='random', svd='numpy_svd'
             message = 'Got svd={}. However, for the current backend ({}), the possible choices are {}'.format(
                     svd, tl.get_backend(), tl.SVD_FUNS)
             raise ValueError(message)
-        
         padded_tensor = _pad_by_zeros(tensor_slices)
-        A = svd_fun(unfold(padded_tensor, 0), n_eigenvecs=rank)[0]
+        A = T.ones((padded_tensor.shape[0], rank), **context)
+
+        unfolded_mode_2 = unfold(padded_tensor, 2)
+        if T.shape(unfolded_mode_2)[0] < rank:
+            raise ValueError("Cannot perform SVD init if rank ({}) is less than the number of columns in each tensor slice ({})".format(
```

Comment:
```suggestion
            raise ValueError("Cannot perform SVD init if rank ({}) is greater than the number of columns in each tensor slice ({})".format(
```

Tiny issue—it appears the message here is backwards from the if statement immediately preceding it.

Reply:
Thanks for spotting @aarmey ! I accepted the change @MarieRoald to trigger the tests.